sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -331,13 +332,18 @@ private static Exception instantiateUnexpectedException(final UnexpectedExceptio
 
         Exception result;
         try {
+            /*
+             * Could we cache this constructor?
+             */
             final Constructor<? extends HttpResponseException> exceptionConstructor =
                 exception.getExceptionType().getConstructor(String.class, HttpResponse.class,
                     exception.getExceptionBodyType());
             result = exceptionConstructor.newInstance("Status code " + responseStatusCode + ", " + bodyRepresentation,
-                httpResponse,
-                responseDecodedContent);
+                httpResponse, responseDecodedContent);
         } catch (ReflectiveOperationException e) {
+            /*
+             * Would this be worth using a StringBuilder?
+             */
```

Comment:
My general thinking here is that we should focus on optimising the critical path. If exceptions are on the critical path, we should optimise, but I hope that these are rarely being hit outside of unit tests, and therefore their performance implications are less important?